brat/Brat/Checker.hs

```diff
@@ -614,14 +614,23 @@ checkBody :: (CheckConstraints m UVerb, EvMode m, ?my :: Modey m)
           -> Checking Src
 checkBody fnName body cty = case body of
   NoLhs tm -> do
-    ((src, _), _) <- makeBox (fnName ++ ".box") cty $ \(overs, unders) -> check tm (overs, unders)
+    ((src, _), _) <- makeBox (fnName ++ ".box") cty $ \conns -> do
+      (((), ()), leftovers) <- check tm conns
+      checkConnectorsUsed (fcOf tm, fcOf tm) (show tm) conns leftovers
     pure src
   Clauses (c :| cs) -> do
     fc <- req AskFC
-    ((box, _), _) <- makeBox (fnName ++ ".box") cty (check (WC fc (Lambda c cs)))
+    ((box, _), _) <- makeBox (fnName ++ ".box") cty $ \conns -> do
+      (((), ()), leftovers) <- check (WC fc (Lambda c cs)) conns
+      checkConnectorsUsed (fcOf (fst c), fcOf (snd c)) (show c) conns leftovers
     pure box
   Undefined -> err (InternalError "Checking undefined clause")
-
+ where
+  checkConnectorsUsed _ _ _ ([], []) = pure ()
+  checkConnectorsUsed (_, tmFC) tm (_, unders) ([], rightUnders) = localFC tmFC $
+    err (TypeMismatch tm (showRow unders) (showRow (filter ((`notElem` (fst <$> rightUnders)) . fst) unders)))
```

Comment:
Can't you do `drop (length rightUnders) unders` instead of the filter? The unused unders should always be on the right?

Reply:
We actually want to `drop` everything _until_ `rightUnders`, it's quite annoying

Reply:
the idiomatic ways i thought of to do this all involved a double reverse, which seems a bit objectionable

Reply:
So `drop (length unders - length rightUnders) unders` also doesn't work?

Reply:
🤦 yes, that should work